### DISCUSSION TITLE
🏈 Remastered: RC3

### DISCUSSION THREAD

**Discussion by swiftyfinch** [swiftyfinch/Rugby#226] [Releases]
You can find the previous release [here](https://github.com/swiftyfinch/Rugby/releases/tag/2.0.0b7).
Use this command to update Rugby from `2.0.0b2+`:
```
rugby update --beta
```
Please, check out the 📖 [readme](https://github.com/swiftyfinch/Rugby/tree/2.0.0b8#readme), you can find there all information.

## Changelog

- 🐛 Fixed `missing resource bundle of dynamic frameworks` errors;
- ⛳️ Removed experimental debug flags. Now `--strip` works as it was in the Rugby1.x;
- 🏛️ Refactored concurrent operations and files managment.

## Release Roadmap 🎯

- [x] RC3
- [ ] **`Release`** 🚀
- [ ] Open Source
    - [ ] Refactoring
    - [ ] Tests
    - [ ] CI Pipelines

## Demo 📺
<img src="https://user-images.githubusercontent.com/64660122/209442704-3f636690-0340-46b5-a3b9-1d9e3779ba82.gif" width=400>

<hr /><em>This discussion was created from the release <a href='https://github.com/swiftyfinch/Rugby/releases/tag/2.0.0b8'>🏈 Remastered: RC3</a>.</em>

**Comment by VaslD** [comment]
Does this version fix an issue where Rugby prints "Swift 5.7" (from one installation of Xcode) but it actually uses Swift 5.7.2 (another installation of Xcode, the active `xcode-select` toolchain) for build and cache management?

I've encountered this several times during Beta and RC but I have yet to find a stable reproduction. It this issue wasn't known I'll see if it happens and get logs uploaded today.

**Reply by swiftyfinch** [reply]
Hello @VaslD! No, I haven't got a such issue yet.
I use `xcodebuild -version` command for getting Xcode version:
```sh
> xcodebuild -version
Xcode 14.1
Build version 14B47b
```

**Reply by swiftyfinch** [reply]
And this one for getting Swift version:
```sh
> swift --version
swift-driver version: 1.62.15 Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

**Reply by swiftyfinch** [reply]
Oh, I see, that because I use the regex like this:
https://github.com/swiftyfinch/Rugby/blob/84072846cbdb04c4137c9384cdc350333ec7ff97/Sources/Rugby/Common/Utils/SwiftVersionProvider.swift#L13

But what's the problem in such implementation? I've just thought that we can skip the patch version of Swift for better cache reuse. Did it break something for you?

**Reply by VaslD** [reply]
Well, it did. Do you use this string only for printing, or do you use it for cache management (i.e. the YAML/hash you mentioned) as well?

Sadly, *module compiled with Swift 5.7 cannot be imported by the Swift 5.7.2 compiler*. I just can't get Xcode to spit this out today for some reason, but if you Google `"module compiled with swift 5.7 cannot be imported by the swift"` (with quoted) you'll see Swift 5.7 is incompatible with neither 5.7.1 nor 5.7.2.

I'll post a screenshot next time I see it. It happens during Rugby build with partial local cache.

P.S.: I don't know if it's impossible (unsupported) to reuse artifacts between Xcode versions, or just because minor versions in 5.7 branch differ too much. After all, Xcode 14.0 (Swift 5.7) and 14.1 (5.7.1) don't even contain the same macOS SDK (major version difference), and this never happened before. Apple surely messed up their release schedule during COVID.

**Reply by swiftyfinch** [reply]
I use the Swift version in the hash for each module like this:
```yml
swift_version: '5.7'
```
Thank you for your detailed answer and often feedback 👍🏻
I'll change the regex and include the Swift patch version in the module hash in the next version.
